spec/requests/movies_spec.rb

```diff
@@ -0,0 +1,337 @@
+      end
+
+      it 'returns redirect status' do
+        get('/movies/new')
+        expect(response.status).to eq(302)
+      end
+    end
+
+    context 'when user without permission' do
+      before { sign_in user }
+      it 'redirects to root' do
+        get('/movies/new')
+        expect(response).to redirect_to('/')
+      end
+
+      it 'returns redirect status' do
+        get('/movies/new')
+        expect(response.status).to eq(302)
+      end
+    end
+
+    context 'when user with permission' do
+      before { sign_in manager }
+      it 'returns successful response' do
+        get('/movies/new')
+        expect(response.status).to eq(200)
+      end
+
+      it 'renders proper template' do
+        get('/movies/new')
+        expect(response).to render_template('movies/new')
+      end
+    end
+  end
+
+  describe 'POST /movies' do
+    let(:params) do
+      {
+        movie: {
+          title:,
+          description: Faker::Movie.quote,
+          duration:
+        }
+      }
+    end
+
+    let(:title) { Faker::Movie.title }
+    let(:duration) { Faker::Number.number(digits: 2).to_i }
+
+    context 'when user not signed in' do
+      it 'redirects to sign_in' do
+        post('/movies')
+        expect(response).to redirect_to('/users/sign_in')
+      end
+
+      it 'returns redirect status' do
+        post('/movies')
+        expect(response.status).to eq(302)
+      end
+
+      it 'does not create a movie' do
+        expect { post('/movies', params:) }.not_to change(Movie, :count)
+      end
+    end
+
+    context 'when user without permission' do
+      before { sign_in user }
+      it 'redirects to root' do
+        post('/movies', params:)
+        expect(response).to redirect_to('/')
+      end
+
+      it 'returns redirect status' do
+        post('/movies')
+        expect(response.status).to eq(302)
+      end
+
+      it 'does not create a movie' do
+        expect { post('/movies', params:) }.not_to change(Movie, :count)
+      end
+    end
+
+    context 'when user with permission' do
+      before { sign_in manager }
+      it 'returns redirect status' do
+        post('/movies', params:)
+        expect(response.status).to eq(302)
+      end
+
+      it 'redirects to movie/movie_id' do
+        post('/movies', params:)
+        expect(response).to redirect_to action: :show, id: assigns(:movie).id
+      end
+
+      it 'creates movie' do
+        expect { post('/movies', params:) }.to change(Movie, :count).by(1)
+      end
+    end
+
+    context 'when invalid params' do
+      before { sign_in manager }
+      let(:duration) { 'some string' }
+
+      it "doesn't create movie" do
+        expect { post('/movies', params:) }.not_to change(Movie, :count)
+      end
+
+      it 'returns unsuccessful response' do
+        post('/movies', params:)
+        expect(response.status).to eq(422)
+      end
+
+      let(:duration) { Faker::Number.number(digits: 2).to_i }
```

Comment:
this is the same definition of duration as above, you don't need to define it again, you have access to it